sdk/dotnet/Pulumi/Deployment/DeploymentOutputs.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections.Generic;
+
+namespace Pulumi
+{
+    /// <summary>
+    /// A type-safe wrapper around <see cref="IDictionary{String, Object}"/> for generating stack outputs from created resources. 
+    /// </summary>
+    public class DeploymentOutputs
+    {
+        private readonly Dictionary<string, object?> outputs;
+
+        private DeploymentOutputs()
+        {
+            outputs = new Dictionary<string, object?>();
```

Comment:
So what's the main advantage of this class over a dictionary? Preventing users from passing objects as values? Would it be better to actually fix that problem and allow passing (a subset of) arbitrary classes?

Reply:
The main advantage is syntactic: make writing the programs more beginner-friendly. Not having to define anything to get going just like python/go/typescript programs

I know we can just say "Look at the types and you will know what to do!" but I believe that if a beginner looks at this
```cs
RunAsync(Func<IDictionary<string, object>>)
```
That 
 - they will be confused if they don't know what `Func<...>` is
 - will probably try to `new IDictionary<...>` and get a compile error
 - will wonder why the values are just any `object` where we have explicit `Output<T>` that come from created resources. 

There is also `RunAsync(Action)` which isn't as useful because it doesn't return outputs 

IMO this is the version that people would actually want to use instead of defining `Stack`s